connectors/sources/network_drive.py

```diff
@@ -200,9 +306,20 @@ async def get_docs(self, filtering=None):
         Yields:
             dictionary: Dictionary containing the Network Drive files and folders as documents
         """
-        directory_details = smbclient.walk(top=rf"\\{self.server_ip}/{self.drive_path}")
 
-        for path, _, _ in directory_details:
+        if filtering and filtering.has_advanced_rules():
+            advanced_rules = filtering.get_advanced_rules()
+            try:
+                matched_paths, _ = self.find_matching_paths(advanced_rules)
+            except Exception as exception:
+                self._logger.warning(
+                    f"Something went wrong while running advanced rules. Exception: {exception}"
+                )
+                raise
```

Comment:
I don't think it will raise exception
```suggestion
            matched_paths, invalid_rules = self.find_matching_paths(advanced_rules)
            if invalid_rules:
                self._logger.error(
                    f"Invalid advanced rules are found: {invalid_rules}"
                )
                raise InvalidRules(...)
```